src/components/PupilComponents/LessonEngineProvider/LessonEngineProvider.tsx

```diff
@@ -9,34 +16,127 @@ export const lessonSections = [
   "review",
 ] as const;
 
-export const lessonReviewSections = [
+export const allLessonReviewSections = [
   "intro",
   "starter-quiz",
   "video",
   "exit-quiz",
 ] as const;
 
 export type LessonSection = (typeof lessonSections)[number];
+export type LessonReviewSection = (typeof allLessonReviewSections)[number];
 
 export const isLessonSection = (
   currentSection: string,
 ): currentSection is LessonSection => {
   return lessonSections.includes(currentSection as LessonSection);
 };
 
-export type LessonSectionResult = { grade: number; numQuestions: number };
+export function isLessonReviewSection(
+  section: string,
+): section is LessonReviewSection {
+  return lessonSections.includes(section as LessonReviewSection);
+}
+
+type QuizResult = { grade: number; numQuestions: number };
+type LessonSectionState = {
+  isComplete: boolean;
```

Comment:
Preferring a boolean next to the section state over maintaining a separate array — easier to query.